HIPs/firefly.md

```diff
@@ -0,0 +1,374 @@
+
+Because there is no existing OSS project that solves these problems, developers are left to themselves to create a solution. These solutions likely end up becoming specialized or tightly coupled to the project they were designed as a part of, and therefore are not reusable. Firefly seeks to fill this gap with a platform that is reusable for any multiparty business transaction backed by blockchain.
+
+## Status
+
+Proposal
+
+## Solution
+
+In a FireFly network, each member of the network deploys and runs one or more FireFly Nodes. Each Node consists of a FireFly Core application, as well as its dependencies. Each dependency is abstracted from the Core by a modular plugin system, so underlying technologies, such as the database for example, can be changed if necessary.
+
+![Multiparty System](../assets/multiparty-system.svg "Multiparty System")
+
+FireFly Core applications register themselves in a Registry, which allows other Nodes to discover them. Each Node has a series of Connectors and Common Utilities that all use plugins to connect them to the Core.
+
+![FireFly Node](../assets/firefly-node.svg "FireFly Node")
+
+## Use Cases
+
+There many hundreds of use cases which analysts have mapped out across all major industries, here are several illustrative examples:
+
+- Insurance - first notice of loss, subrogation, re-insurance document flows
+- Healthcare - coordination of care, provider data quality, clinical trial data management
+- Financial Services - digital asset management, bill of lading, certificate of indemnity
+- Supply Chain - track and trace, food safety, supplier payments
+
+These are obviously a very small sample of the use case population. When we’ve worked across these industries and others, we’ve noticed that all of these use cases follow remarkably similar patterns:
+
+- Multi-party business process
+- Digital twin
+- Digital Asset platform (STO lifecycle management)
+- Document-based business transactions
+- Data marketplace or sharing
+- Chain of custody
+- Delivery vs. payment
+
+We can further decompose into a handful of common operations:
+
+- **Blockchain-backed operations**
+  - On-chain/off-chain event aggregation
+  - Sequenced off-chain logic execution
+  - Broadcast + multi-response
+  - Broadcast + single-response
+  - Private document transfer
+  - Off-chain private or group exchange
+- **Blockchain-based operations**
+  - Orchestration of group logic execution
+  - Digital asset lifecycle management
+  - On-chain private or group exchange
+- **Hybrid (both blockchain-backed and blockchain-based)**
+  - Delivery vs. payment
+  - On-chain authorization of off-chain coordinated actions
+
+The FireFly core exposes a simple set of API’s to accomplish these common operations. All of the plumbing to reliably execute these operations is included in the FireFly node.
+
+## Sample Scenario and how it works in FireFly
```

Comment:
Later, it is claimed that FireFly has `Data privacy by default`. Can you describe in details what it means in the context of this scenario?

Reply:
This question should be addressed by the clarifications added after the sequence diagram.